File: src/#24/solution.js
function ListNode(val, next) {
  this.val = val === undefined ? 0 : val
  this.next = next === undefined ? null : next
}

var swapPairs = function (head) {
  if (head === null) {
    return null
  }
  if (head.next === null) {
    return head
  }
  let prev = new ListNode(0, null)
  let current = head
  head = head.next

  while (current.next !== null && current.next.next !== null) {
    if (current.next !== null) {
      let temp = current.next
      current.next = temp.next
      temp.next = current
      prev.next = temp
      prev = current
      current = current.next
    }
  }
  if (current.next !== null) {
    let temp = current.next
    current.next = temp.next
    temp.next = current
    prev.next = temp
  }

  return head
}

File: src/#24/solution.go
package solution

type ListNode struct {
	Val  int
	Next *ListNode
}

func swapPairs(head *ListNode) *ListNode {
	if head == nil {
		return nil
	}
	if head.Next == nil {
		return head
	}
	prev := &ListNode{0, nil}
	current := head
	head = head.Next
	for current.Next != nil && current.Next.Next != nil {
		if current.Next != nil {
			temp := current.Next
			current.Next = temp.Next
			temp.Next = current

			prev.Next = temp
			prev = current
			
			current = current.Next
		}
	}
	if current.Next != nil {
		temp := current.Next
		current.Next = temp.Next
		temp.Next = current
		prev.Next = temp
	}
	return head
}
